### PR TITLE
Extend option to run tests selectively to slow, unit, integration

### PR DESCRIPTION
**Pull request by n-hutton**
This should now be possible:

```
./scripts/ci-tool.py -T Debug --test-names chain-unit-tests,ledger-chain-tests
```

### CODE REVIEW

File: scripts/ci-tool.py [hunk 1]
```diff
@@ -249,6 +249,12 @@ def test_project(build_root, include_regex=None, exclude_regex=None):
         '-T', TEST_NAME
     ]
 
+    # Need to convert test names from set to a regex of ORs
+    names_as_list = list(test_names)
+    names_as_regex = "|".join(str(x) for x in names_as_list)
```
Comment (wkaluza) [inline scripts/ci-tool.py]: You don't need the list here, do you? You can iterate over `test_names` directly.

File: scripts/end_to_end_test/run_end_to_end_test.py [hunk 2]
```diff
@@ -620,20 +620,23 @@ def run_test(build_directory, yaml_file, node_exe, name_filter=None):
             for test in all_yaml:
                 # Get test setup conditions
                 setup_conditions = yaml_extract(test, 'setup_conditions')
+                description = yaml_extract(test, 'test_description')
+
+                test_disabled = "DISABLED" in description
 
-                # Check if name is not filtered out
+                # Check if name is not filtered out (overrides disabled specifier)
```
Comment (wkaluza) [inline scripts/end_to_end_test/run_end_to_end_test.py]: I don't think this is a good idea. If someone is working on a test, they might as well re-enable it in the YAML. This creates a risk where someone will locally fix a disabled test and commit changes without removing the `DISABLED` tag. This mistake would not be obvious from the GitHub diff and thus would be easy to miss in review.

**Review by wkaluza — CHANGES_REQUESTED** [review]
Good change, but I wouldn't alter the disablement logic in e2e tests.

File: scripts/ci-tool.py (hunk 1 above)
Comment (n-hutton) [reply scripts/ci-tool.py]: you're right

File: scripts/end_to_end_test/run_end_to_end_test.py (hunk 2 above)
Comment (n-hutton) [reply scripts/end_to_end_test/run_end_to_end_test.py]: I can see this happening and I guess it is not much effort to enable the test

**Review by wkaluza — APPROVED** [review]

**Pull request merged**